lib/health-indicator/database/mikro-orm.health.ts

```diff
@@ -0,0 +1,141 @@
+import { ModuleRef } from '@nestjs/core';
+import { HealthCheckError } from '../../health-check/health-check.error';
+
+import * as MikroOrmNestJS from '@mikro-orm/core';
+import { TimeoutError } from '../../errors';
+import {
+  TimeoutError as PromiseTimeoutError,
+  promiseTimeout,
+  checkPackages,
+} from '../../utils';
+import { HealthIndicator, HealthIndicatorResult } from '..';
+
+export interface MikroOrmPingCheckSettings {
+  /**
+   * The connection which the ping check should get executed
+   */
+  connection?: any;
+  /**
+   * The amount of time the check should require in ms
+   */
+  timeout?: number;
+}
+
+/**
+ * The MikroOrmHealthIndicator contains health indicators
+ * which are used for health checks related to MikroOrm
+ *
+ * @publicApi
+ * @module TerminusModule
+ */
+@Injectable({ scope: Scope.TRANSIENT })
+export class MikroOrmHealthIndicator extends HealthIndicator {
+  /**
+   * Initializes the MikroOrmHealthIndicator
+   *
+   * @param {ModuleRef} moduleRef The NestJS module reference
+   */
+  constructor(private moduleRef: ModuleRef) {
+    super();
+    this.checkDependantPackages();
+  }
+
+  /**
+   * Checks if responds in (default) 1000ms and
+   * returns a result object corresponding to the result
+   * @param key The key which will be used for the result object
+   * @param options The options for the ping
+   *
+   * @example
+   * MikroOrmHealthIndicator.pingCheck('database', { timeout: 1500 });
+   */
+  public async pingCheck(
+    key: string,
+    options: MikroOrmPingCheckSettings = {},
+  ): Promise<HealthIndicatorResult> {
+    this.checkDependantPackages();
+
+    const connection = options.connection || this.getContextConnection();
+    const timeout = options.timeout || 1000;
+
+    if (!connection) {
+      return this.getStatus(key, false);
+    }
+
+    try {
+      await this.pingDb(connection, timeout);
+    } catch (error) {
+      // Check if the error is a timeout error
+      if (error instanceof PromiseTimeoutError) {
+        throw new TimeoutError(
+          timeout,
+          this.getStatus(key, false, {
+            message: `timeout of ${timeout}ms exceeded`,
+          }),
+        );
+      }
+
+      // Check if the error is a connection not found error
+      if (error instanceof Error) {
+        throw new HealthCheckError(
+          error.message,
+          this.getStatus(key, false, {
+            message: error.message,
+          }),
+        );
+      }
+    }
+
+    return this.getStatus(key, true);
+  }
+
+  private checkDependantPackages() {
+    checkPackages(
+      ['@mikro-orm/nestjs', '@mikro-orm/core'],
+      this.constructor.name,
+    );
+  }
+
+  /**
+   * Returns the connection of the current DI context
+   */
+  private getContextConnection(): MikroOrmNestJS.Connection | null {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { MikroORM } = require('@mikro-orm/nestjs') as typeof MikroOrmNestJS;
```

Comment:
@BrunnerLivio  this should be `@mikro-orm/core`

`const { MikroORM } = require('@mikro-orm/core') as typeof MikroOrmNestJS;`

e2e tests pass locally with this change.

Reply:
Oh okay, let's try this then :) 

```suggestion
    const { MikroORM } = require('@mikro-orm/core') as typeof MikroOrmNestJS;
```

Reply:
Awesome! Thanks a lot, that seems to have done the trick. Currently the e2e tests seem to fail because of a different change I did, so I will revert that tonight and see that we can get this released :)